vscode-react/src/components/deployments/organisms/executePopper.component.tsx

```diff
@@ -1,89 +1,199 @@
 	displayedErrors,
-}) => {
+}: ExecutePopperProps) => {
+	const [inputParameters, setInputParameters] = useState<InputParameter[]>([]);
+	const [hasValidationErrors, setHasValidationErrors] = useState(false);
+
+	const validateParameters = () => {
+		const updatedParams = inputParameters.map((param) => ({
+			...param,
+			keyError: param.key.length === 0 ? "Key must not be empty" : undefined,
+			valueError: param.value.length === 0 ? "Value must not be empty" : undefined,
+		}));
+		setInputParameters(updatedParams);
+		return updatedParams.every((param) => !param.keyError && !param.valueError);
+	};
+
+	useEffect(() => {
+		setHasValidationErrors(inputParameters.some((param) => param.keyError || param.valueError));
+	}, [inputParameters]);
+
 	const onStartClick = (event: MouseEvent<HTMLElement> | undefined) => {
 		event?.stopPropagation();
-		onStartSession();
+		if (validateParameters()) {
+			onStartSession(inputParameters.map(({ key, value }) => ({ key, value })));
+		}
 	};
 
 	const onFileChangeClick = (event: any): void => {
 		event.stopPropagation();
 		onFileChange(event.target.value);
 	};
 
-	const onEntrypointClick = (event: any): void => {
+	const onFunctionNameChange = (event: any): void => {
 		event.stopPropagation();
 		onFunctionChange(event.target.value);
 	};
 
+	const addParameter = () => {
+		setInputParameters([...inputParameters, { key: "", value: "" }]);
+	};
+
+	const removeParameter = (index: number) => {
+		// eslint-disable-next-line @typescript-eslint/naming-convention
+		const updatedParams = inputParameters.filter((_, i) => i !== index);
+		setInputParameters(updatedParams);
+	};
+
+	const updateParameter = (index: number, field: "key" | "value", value: string) => {
+		const updatedParams = inputParameters.map((param, i) => {
+			if (i === index) {
+				const updatedParam = { ...param, [field]: value };
+				if (field === "key") {
+					updatedParam.keyError =
+						value.length === 0 ? translate().t("reactApp.deployments.inputParameters.errors.missingKey") : undefined;
+				} else {
+					updatedParam.valueError =
+						value.length === 0 ? translate().t("reactApp.deployments.inputParameters.errors.missingValue") : undefined;
+				}
+				return updatedParam;
+			}
+			return param;
+		});
+		setInputParameters(updatedParams);
+	};
+
+	const fullWidthStyle = {
+		// eslint-disable-next-line @typescript-eslint/naming-convention
+		"--vscode-inputbox-width": "100%",
+	} as React.CSSProperties;
+
+	useEffect(() => {
+		const styleElement = document.createElement("style");
+		styleElement.textContent = `
+			vscode-text-field, vscode-dropdown {
+			width: 100% !important;
+			}
+			vscode-text-field::part(root), vscode-dropdown::part(root) {
+			width: 100% !important;
+			}
+			vscode-text-field::part(control), vscode-dropdown::part(control) {
+			width: 100% !important;
+			}
+		`;
+		document.head.appendChild(styleElement);
+
+		return () => {
+			document.head.removeChild(styleElement);
+		};
+	}, []);
```

Comment:
remove

Reply:
It's a must, I can't remove it, otherwise, the width doesn't work, because the way vscode extensions work.